packages/opencensus-web-instrumentation-zone/src/interaction-tracker.ts

```diff
@@ -29,39 +31,93 @@ export type AsyncTask = Task & {
 };
 
 export class InteractionTracker {
+  private readonly tracingZones: { [index: string]: Zone } = {};
   constructor() {
+    const interactionTracker: InteractionTracker = this;
+
     const runTask = Zone.prototype.runTask;
     Zone.prototype.runTask = function(
       task: AsyncTask,
       applyThis: unknown,
       applyArgs: unknown
     ) {
-      if (task.eventName && task.eventName.toString().indexOf('click') !== -1) {
-        console.log('Running task');
+      const time = Date.now();
+
+      console.warn('Running task');
+      console.log(task.zone);
+
+      let taskZone = this;
+      if (isTrackedElement(task)) {
         console.log('Click detected');
+
+        const zoneName = randomSpanId();
+        const tracingZone = Zone.root.fork({
+          name: zoneName,
+          properties: {
+            isTracingZone: true,
+            tracingId: randomTraceId(),
+          },
+        });
+
+        interactionTracker.tracingZones[zoneName] = tracingZone;
+
+        // Change the zone task.
+        // tslint:disable:no-any
+        (task as any)._zone = tracingZone;
```

Comment:
Rather than using `any` here, could you create a new interface something like this:
```
/** Interface that allows access to the private `_zone` property of a Zone.js Task. */
interface TaskWithPrivateZone {
  _zone: Zone;
}
```

Then you can do `(task as TaskWithPrivateZone)._zone = tracingZone`, but with a bit higher type safety and without needing the `tslint:disable:no-any`

Reply:
Instead of creating a new Interface, I added this property in AsyncTask type.